tests/integration/Owncloud/OcisPhpSdk/OcisTest.php

```diff
@@ -535,4 +557,115 @@ public function testGetResourceInvalidId(): void
         $ocis = $this->getOcis('admin', 'admin');
         $ocis->getResourceById('not-existing-id');
     }
+
+    public function testGetShareCreatedByMe(): void
+    {
+        $ocis = $this->getOcis('admin', 'admin');
+        $einstein = $ocis->getUsers('einstein')[0];
+        $personalDrive = $this->getPersonalDrive($ocis);
+        $personalDrive->createFolder('newFolder');
+        $this->createdResources[$personalDrive->getId()][] = '/newFolder';
+        $resources = $personalDrive->getResources();
+        $sharedResource = null;
+        foreach ($resources as $resource) {
+            if ($resource->getName() === 'newFolder') {
+                $sharedResource = $resource;
+            }
+        }
+        if(empty($sharedResource)) {
+            throw new \Error(
+                "resource not found "
+            );
+        }
+        $editorRole = null;
+        foreach($sharedResource->getRoles() as $role) {
+            if($role->getDisplayName() === 'Editor') {
+                $editorRole = $role;
+            }
+        }
+        if(empty($editorRole)) {
+            throw new \Error(
+                "Editor role not found "
+            );
+        }
+        $sharedResource->invite([$einstein], $editorRole);
+        $myShare = $ocis->getSharedByMe();
+        $this->assertInstanceOf(ShareCreated::class, $myShare[0]);
+        $this->assertEquals($myShare[0]->getReceiver()->getDisplayName(), 'Albert Einstein');
+    }
+
+    public function testGetShareCreatedAndShareCreatedLinkByMe(): void
+    {
+        $ocis = $this->getOcis('admin', 'admin');
+        $einstein = $ocis->getUsers('einstein')[0];
+        $personalDrive = $this->getPersonalDrive($ocis);
+        $personalDrive->createFolder('newFolder');
+        $this->createdResources[$personalDrive->getId()][] = '/newFolder';
+        $resources = $personalDrive->getResources();
+        $sharedResource = null;
+        foreach ($resources as $resource) {
+            if ($resource->getName() === 'newFolder') {
+                $sharedResource = $resource;
+            }
+        }
+        if(empty($sharedResource)) {
+            throw new \Error(
+                "resource not found "
+            );
+        }
+        $editorRole = null;
+        foreach($sharedResource->getRoles() as $role) {
+            if($role->getDisplayName() === 'Editor') {
+                $editorRole = $role;
+            }
+        }
+        if(empty($editorRole)) {
+            throw new \Error(
+                "Editor role not found "
+            );
+        }
+        $sharedResource->invite([$einstein], $editorRole);
+        $sharedResource->createSharingLink(SharingLinkType::VIEW, new \DateTimeImmutable('2023-12-31 01:02:03.456789'), 'Test@123', '');
+        $myShare = $ocis->getSharedByMe();
+        $this->assertInstanceOf(ShareCreated::class, $myShare[0]);
+        $this->assertInstanceOf(ShareLink::class, $myShare[1]);
```

Comment:
Is the order guaranteed ?

Reply:
I've tested and found that the order remains `ShareCreated` and `ShareLink`.